src/components/MusicList.vue

```diff
@@ -0,0 +1,156 @@
+      <p>There is no music to show.</p>
+    </div>
+    <md-dialog v-bind:md-active.sync="addMusicDialogActive">
+      <md-dialog-title>Add Music</md-dialog-title>
+      <md-dialog-content>
+        <form novalidate class="md-layout">
+          <md-field>
+            <label for="dialog-music-name">Music Name</label>
+            <md-input name="dialog-music-name" id="dialog-music-name" v-model="newMusic.music_name"/>
+          </md-field>
+          <md-field>
+            <label for="dialog-artist-id">Artist ID</label>
+            <md-input name="dialog-artist-id" id="dialog-artist-id" type="number" v-model="newMusic.artist_id"/>
+          </md-field>
+          <md-field>
+            <label for="dialog-difficulty">Difficulty</label>
+            <md-input name="dialog-difficulty" id="dialog-difficulty" v-model="newMusic.difficulty"/>
+          </md-field>
+        </form>
+      </md-dialog-content>
+      <md-dialog-actions style="margin: 1rem;">
+        <button class="md-button" v-on:click="addMusicDialogActive = false">
+          <md-icon>close</md-icon> &nbsp; <span>Cancel</span>
+        </button>
+        <button class="md-button" v-on:click="addMusic">
+          <md-icon>check</md-icon> &nbsp; <span>Submit</span>
+        </button>
+      </md-dialog-actions>
+    </md-dialog>
+    <md-dialog-confirm
+      md-title="Delete Music?"
+      v-bind:md-content="'Are you sure you want to delete music ' + deleteMusicName + '?'"
+      md-confirm-text="Yes"
+      md-cancel-text="No"
+      v-bind:md-active.sync="deleteMusicDialogActive"
+      @md-cancel="deleteMusicDialogActive = false"
+      @md-confirm="doDeleteMusic" />
+  </div>
+</template>
+
+<script>
+import MusicInfo from "@/components/MusicInfo";
+
+export default {
+  name: "MusicList",
+  components: {
+    'music-info': MusicInfo
+  },
+  data: function () {
+    return {
+      loading: true,
+      loadingMessage: 'Loading...',
+      musics: [],
+      addMusicDialogActive: false,
+      newMusic: {
+        music_name: '',
+        artist_id: 0,
+        difficulty: ''
+      },
+      deleteMusicDialogActive: false,
+      deleteMusicIndex: 0,
+      deleteMusicName: ''
+    }
+  },
+  created: function () {
+    this.getMusics();
+  },
+  methods: {
+    getMusics() {
+      this.$http.get('/show')
+          .then(res => {
+            this.musics = res.data;
+            this.loading = false;
+          })
+          .catch(err => {
+            console.error(err);
+            this.loadingMessage = err;
+          });
+    },
+    addMusic() {
+      this.$http.post('/add', this.newMusic)
+          .then(() => {
+            this.getMusics();
+            this.addMusicDialogActive = false;
+          })
+          .catch(err => console.error(err));
+    },
+    deleteMusic(index) {
+      this.deleteMusicIndex = index;
+      this.deleteMusicName = this.musics[index].music_name;
+      this.deleteMusicDialogActive = true;
+    },
+    doDeleteMusic() {
```

Comment:
For consistency with playMusic, restoreMusic, etc., maybe deleteMusic can have the post method, and doDeleteMusic can perform current "deleteMusic"'s job.

Also, could you comment the relationship between deleteMusic an doDeleteMusic?

Reply:
When user click on 'delete' button, the app will pop up a dialog to let user confirm whether he/she really wants to delete the music. `deleteMusic` will open the dialog and set the index of music data, and `doDeleteMusic` will make the API call to delete data stored in server. I'll discuss with team members on improving this part of code this week too. 😄

Reply:
I see, then the original naming makes sense, thanks for the explanation. Could you comment this explanation in code?

Reply:
Comments are added in 421fbc2.